core/src/labels/labels.cpp

```diff
@@ -60,13 +60,14 @@ void Labels::processLabelUpdate(const ViewState& viewState,
             m_needUpdate |= label->evalState(dt);
             label->addVerticesToMesh();
         }
+        m_allLabels.emplace_back(label.get(), tile, isProxy);
```

Comment:
Yep that's a good point! @hjanetzek also proposed it, I will clean up that branch since it's a lot of debug code and will update for that.